src/m365/planner/commands/task/task-checklistitem-remove.ts

```diff
@@ -0,0 +1,124 @@
+import { Cli, Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: string;
+  taskId: string;
+  confirm?: boolean;
+}
+
+class PlannerTaskChecklistItemRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return commands.TASK_CHECKLISTITEM_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the checklist item from the planner task';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'title', 'isChecked'];
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.confirm = (!(!args.options.confirm)).toString();
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (args.options.confirm) {
+      this.removeChecklistitem(logger, args, cb);
+    }
+    else {
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the checklist item with id ${args.options.id} from the planner task?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          this.removeChecklistitem(logger, args, cb);
+        }
+      });
+    }
+  }
+
+  private removeChecklistitem(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    this
+      .getTaskDetailsEtag(args.options.taskId)
+      .then(etag => {
+        const requestOptionsTaskDetails: any = {
+          url: `${this.resource}/v1.0/planner/tasks/${args.options.taskId}/details`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none',
+            'If-Match': etag,
+            'Prefer': 'return=representation'
+          },
+          responseType: 'json',
+          data: {
+            checklist: {
+              [args.options.id]: null
+            }
+          }
+        };
+
+        return request.patch(requestOptionsTaskDetails);
+      })
+      .then((res: any): void => {
+        if (!args.options.output || args.options.output === 'json') {
```

Comment:
On a succesful remove, just call `cb()`. Thats all you need. No logging necessary.

Reply:
Thank you. I have implemented the changes. Can you please review?